### PR TITLE
6.get comments by article

### PR DESCRIPTION
**Pull request by SamirabintABS**

### CODE REVIEW

**Comment by Harrison365** [comment]
Hi Samira :) 

This is looking good so far. Please could you take a look at the following: 
- Your response should be on a key of comments.
- The article_id you are testing for should be specific (e.g. 3 rather than expect.any(Number).
- We'd like another 200 test for an article that exists but that has no comments. 

Please let us know when you want us to take another look 👍 Thanks

File: __tests__/app.test.js [hunk 1]
```diff
@@ -104,4 +104,60 @@ describe('GET: /api/articles', () => {
...
+});
+
+describe('GET /api/articles/:article_id/comments', () => {
+    it('GET 200: responds with an array of comment objects for the given article_id with the relevant properties, sorted by the most recent comments first', () => {
+        return request(app)
+            .get('/api/articles/1/comments')
+            .expect(200)
+            .then(({ body }) => {
+                const { comments } = body;
+                console.log(comments)
```
Comment (popatre) [inline __tests__/app.test.js]: remove log please

File: __tests__/app.test.js [hunk 2]
```diff
@@ -104,4 +104,60 @@ describe('GET: /api/articles', () => {
...
+                const { rows } = comments;
```
Comment (popatre) [inline __tests__/app.test.js]: This comment key shouldnt have a key of rows inside of it

File: __tests__/app.test.js [hunk 3]
```diff
@@ -104,4 +104,60 @@ describe('GET: /api/articles', () => {
...
+                })
+            });
+    })
+    it('GET 200: responds with an empty array if article exists but has no comments', () => {
+        return request(app)
+            .get('/api/articles/2/comments')
+            .expect(200)
+            .then(({ body }) => {
+                const { comments } = body;
+                const { rows } = comments;
```
Comment (popatre) [inline __tests__/app.test.js]: Good, just needs there to not be a key of rows

File: __tests__/app.test.js [hunk 4]
```diff
@@ -104,4 +104,60 @@ describe('GET: /api/articles', () => {
...
+                expect(rows).toBeInstanceOf(Array);
+                expect(rows).toEqual([]);
+            })
+    });
+    it('GET 404: Responds with an error if the article id is valid but does not exist yet or if the article id exists but has not comments', () => {
+        return request(app)
+            .get("/api/articles/99999/comments")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("No comments found")
```
Comment (popatre) [inline __tests__/app.test.js]: Would be better as `article id not found`

File: models/app.model.js [hunk 5]
```diff
@@ -43,4 +43,19 @@ exports.fetchAllArticles = () => {
...
+    const commentsQuery = `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`
+
+    return Promise.all([
+        db.query(articlesQuery, [articleId]),
+        db.query(commentsQuery, [articleId])
+    ]).then(([articleResult, commentsResult]) => {
+        if (articleResult.rows.length === 0) {
+            return Promise.reject({ status: 404, msg: "No comments found" })
+        }
+        return commentsResult;
```
Comment (popatre) [inline models/app.model.js]: To get rid of the rows key, return `commentsResult.rows`

File: models/app.model.js [hunk 6]
```diff
@@ -43,4 +43,19 @@ exports.fetchAllArticles = () => {
...
         })
-}
\ No newline at end of file
+}
+
+exports.fetchCommentsById = (articleId) => {
+    const articlesQuery = `SELECT * FROM articles WHERE article_id = $1`;
+    const commentsQuery = `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`
+
+    return Promise.all([
```
Comment (popatre) [inline models/app.model.js:52]: Lovely promise.all

File: models/app.model.js [hunk 7]
```diff
@@ -43,4 +43,19 @@ exports.fetchAllArticles = () => {
         .then((result) => {
             return result.rows;
         })
-}
\ No newline at end of file
+}
+
+exports.fetchCommentsById = (articleId) => {
+    const articlesQuery = `SELECT * FROM articles WHERE article_id = $1`;
```
Comment (popatre) [inline models/app.model.js:49]: nicely saved into variables

**Review by popatre — CHANGES_REQUESTED** [review]
Hello, just a few really small changes regarding keys and some error messages. Only small things to fix

**Review by popatre — APPROVED** [review]
Looks good to go, well done 👍

**Pull request merged**